include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1832 @@
+        {}
+
+        //!\brief Returns the value of the referenced bit.
+        constexpr operator bool() const noexcept
+        {
+            return static_cast<bool>(internal.bits & mask);
+        }
+
+        //!\brief Returns the inverted value of the referenced bit.
+        constexpr bool operator~() const noexcept
+        {
+            return !static_cast<bool>(internal.bits & mask);
+        }
+
+        //!\brief Sets the referenced bit to value.
+        constexpr reference_proxy_type & operator=(bool const value)
+        {
+            value ? set() : reset();
+            return *this;
+        }
+
+        //!\brief Sets the referenced bit to the result of a bitwise or with value.
+        constexpr reference_proxy_type & operator|=(bool const value)
+        {
+            if (value)
+                set();
+            return *this;
+        }
+
+        //!\brief Sets the referenced bit to the result of a bitwise and with value.
+        constexpr reference_proxy_type & operator&=(bool const value)
+        {
+            if (!value)
+                reset();
+            return *this;
+        }
+
+        //!\brief Sets the referenced bit to the result of a bitwise xor with value.
+        constexpr reference_proxy_type & operator^=(bool const value)
+        {
+            operator bool() && value ? reset() : set();
+            return *this;
+        }
+
+    private:
+        //!\brief The proxy of the underlying data type.
+        bitfield & internal;
+        //!\brief Bitmask to access one specific bit.
+        uint64_t mask;
+
+        //!\brief Sets the referenced bit to 1.
+        constexpr void set() noexcept
+        {
+            internal.bits |= mask;
+        }
+
+        //!\brief Sets the referenced bit to 0.
+        constexpr void reset() noexcept
+        {
+            internal.bits &= ~mask;
+        }
+    };
+
+public:
+    static_assert(capacity_ <= 58, "The capacity of the dynamic_bitset exceeds the limit of 58.");
+
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief Equals bool.
+    using value_type      = bool;
+    //!\brief A proxy type that enables assignment.
+    using reference       = reference_proxy_type;
+    //!\brief Equals the value_type.
+    using const_reference = bool;
+    //!\brief The iterator type of this container (a random access iterator).
+    using iterator        = detail::random_access_iterator<dynamic_bitset>;
+    //!\brief The const_iterator type of this container (a random access iterator).
+    using const_iterator  = detail::random_access_iterator<dynamic_bitset const>;
+    //!\brief A std::ptrdiff_t.
+    using difference_type = ptrdiff_t;
+    //!\brief An unsigned integer type (usually std::size_t)
+    using size_type       = detail::min_viable_uint_t<capacity_>;
+    //!\}
+
+    //!\cond
+    // this signals to range-v3 that something is a container :|
+    using allocator_type    = void;
```

Comment:
> you want this to be not a view, right? So the correct way per standard is to overload

I want this to be a container recognised by the range library. We added the `allocator_type` for each of our container; otherwise the container_concept of the range library fails. 
I think it doesn't have to do something with views?

Reply:
As far as I can tell, the container concept of the ranges library does not check for a type `allocator_type`. Can you give me an example where the expected behavior does not work without this type?

Reply:
It's used in `range/v3/range/conversion.hpp` as a hack for converting ranges to containers or something?

Idk about an example, I'd rather do a card for the dev meeting since this affects all the containers...